azure/scope/identity.go

```diff
@@ -165,22 +161,14 @@ func (p *AzureCredentialsProvider) GetTokenCredential(ctx context.Context, resou
 		}
 		cred, authErr = NewWorkloadIdentityCredential(azwiCredOptions)
 
-	case infrav1.ServicePrincipal, infrav1.ServicePrincipalCertificate, infrav1.UserAssignedMSI:
-		if err := createAzureIdentityWithBindings(ctx, p.Identity, resourceManagerEndpoint, activeDirectoryEndpoint, clusterMeta, p.Client); err != nil {
-			return nil, err
-		}
-
-		options := azidentity.ManagedIdentityCredentialOptions{
-			ID: azidentity.ClientID(p.Identity.Spec.ClientID),
-		}
-		cred, authErr = azidentity.NewManagedIdentityCredential(&options)
-
 	case infrav1.ManualServicePrincipal:
+		log.Info("Identity type ManualServicePrincipal is deprecated and will be removed in a future release. See https://capz.sigs.k8s.io/topics/workload-identity to get started with Workload Identity instead.")
+		fallthrough
```

Comment:
I think we've now hit keyword bingo

Reply:
Serious question, how does this work? It won't match `case infrav1.ServicePrincipal` so how would it get into the next case?

Reply:
fallthrough emulates the default behavior for C-style switch statements without a `break` where control flow continues to the first line of the next case.

Reply:
Might be the first time I've actually used `fallthrough` in Go. It does look weird.